include/sparrow/layout/array_wrapper.hpp

```diff
@@ -178,23 +178,23 @@ namespace sparrow
 
     template <class T>
     array_wrapper_impl<T>::array_wrapper_impl(T&& ar)
-        : array_wrapper(this->get_data_type())
+        : array_wrapper(detail::array_access::get_arrow_proxy(ar).data_type())
```

Comment:
This modification was made because the fixed_width_binary_array used a std::vector<byte_t> but it's already registered for BINARY: https://github.com/man-group/sparrow/blob/main/include/sparrow/types/data_traits.hpp#L57
The other solution is to use a custom vector type just to avoid that.

Reply:
REMINDER: if we keep my modification: Remove unused code

Reply:
https://github.com/man-group/sparrow/pull/320#issuecomment-2590188236